circuits/core/src/algorithms/merkle_path/to_root.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+use crate::traits::Hash;
+
+impl<E: Environment, H: Hash> MerklePath<E, H>
+where
+    <<H as Hash>::Output as Ternary>::Boolean: From<Boolean<E>>,
+    <H as Hash>::Output: From<<<H as Hash>::Output as Ternary>::Output>,
+    Vec<<H as Hash>::Input>: From<<H as Hash>::Output>,
```

Comment:
Is there a way to use implied trait bounds?

Reply:
Reduced to:
```
<<H as Hash>::Output as Ternary>::Boolean: From<Boolean<E>>,
Vec<<H as Hash>::Input>: From<Vec<<<H as Hash>::Output as ToBits>::Boolean>>,
```

Boolean conversion is a bit trickier because each object has it's own `Boolean: BooleanTrait` type.